data/data/azure/vnet/main.tf

```
@@ -50,6 +50,12 @@ resource "azurerm_storage_account" "cluster" {
   account_replication_type        = "LRS"
   min_tls_version                 = contains(local.environments_with_min_tls_version, var.azure_environment) ? "TLS1_2" : null
   allow_nested_items_to_be_public = false
+
+  network_rules {
+    default_action             = "Allow"
```

I think we want to set this to Deny
```suggestion
    default_action             = "Deny"
```
The [docs](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account#default_action) are a little confusing, but it says:

> If specifying network_rules, one of either ip_rules or virtual_network_subnet_ids must be specified and default_action must be set to Deny.

So this means the default action will be to deny traffic, unless it is in one of the designated subnets.

---

After changing it to Deny, I am running into the following error 
https://github.com/hashicorp/terraform-provider-azurerm/issues/2977

---

> I think we want to set this to Deny
> 
> The [docs](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account#default_action) are a little confusing, but it says:
> 
> > If specifying network_rules, one of either ip_rules or virtual_network_subnet_ids must be specified and default_action must be set to Deny.
> 
> So this means the default action will be to deny traffic, unless it is in one of the designated subnets.

In practice, when setting the default to deny without a whitelist IP, this block will fail. After diving into this, it appears we must either whitelist an IP address or range using ip_rules, or create a private endpoint. Creating a private endpoint requires allocating a private IP address and an associated private DNS record. I'm not sure how realistic that will be for some folks.